superdesk/io/format_document_for_preview.py

```diff
@@ -0,0 +1,43 @@
+from flask import Blueprint, request, Response, current_app as app
+import superdesk
+from superdesk import get_resource_service
+from superdesk.publish.formatters import get_formatter
+from apps.content_types import apply_schema
+
+
+bp = Blueprint('format_document', __name__)
+
+def get_mime_type(formatter_qcode):
+    if(formatter_qcode == 'newsmlg2'):
+        return 'text/xml'
+    else:
+        return 'application/json'
+
+
+@bp.route('/format-document-for-preview/', methods=['GET', 'OPTIONS'])
+def format_document():
+
+    document_id = request.args.get('document_id')
+    subscriber_id = request.args.get('subscriber_id')
+    formatter_qcode = request.args.get('formatter')
+
+    subscriber = get_resource_service('subscribers').find_one(req=None, _id=subscriber_id)
+    doc = get_resource_service('archive').find_one(req=None, _id=document_id)
+
+    formatter = get_formatter(formatter_qcode, doc)
+    formatted_docs = formatter.format(article=apply_schema(doc),
+                                      subscriber=subscriber,
+                                      codes=None) # `codes` doesn't seem to be used in any formatter,
```

Comment:
Can we remove the argument or is it used somewhere?

Reply:
`codes` are used by https://github.com/superdesk/superdesk-aap/blob/master/server/aap/publish/formatters/anpa_formatter.py#L39. 

These `codes` are used by upstream distribution platform at AAP and they are calculated based on the products assigned to the subscriber.

Please don't remove the argument.